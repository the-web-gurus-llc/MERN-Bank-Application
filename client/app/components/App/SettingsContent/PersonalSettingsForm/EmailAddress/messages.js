/*
 * EmailAddress Messages
 *
 * This contains all the text for the EmailAddress component.
 */

import { defineMessages } from 'react-intl';

export const scope = 'app.components.SettingsContent.EmailAddress';

export default defineMessages({
  label: {
    id: `${scope}.label`,
    defaultMessage: 'E-Mail Address',
  },
  validationEmailValid: {
    id: `${scope}.validation.email.vaild`,
    defaultMessage: 'E-Mail address must be valid.',
  },
  placeholder: {
    id: `${scope}.placeholder`,
    defaultMessage: 'Enter new e-mail address',
  },
  emailExist: {
    id: `${scope}.emailExist`,
    defaultMessage: 'E-Mail address already exists.',
  },
});
