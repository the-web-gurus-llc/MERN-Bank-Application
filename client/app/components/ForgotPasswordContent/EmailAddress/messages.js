/*
 * EmailAddressForm Messages
 *
 * This contains all the text for the EmailAddressForm component.
 */

import { defineMessages } from 'react-intl';

export const scope = 'app.components.ForgetPassword.ForgetPasswordContent.EmailAddress';

export default defineMessages({
  label: {
    id: `${scope}.label`,
    defaultMessage: 'E-Mail Address',
  },
  validation_email_valid: {
    id: `${scope}.validation_email_valid`,
    defaultMessage: 'E-Mail address must be valid.',
  },
  validation_email_required: {
    id: `${scope}.validation_email_required`,
    defaultMessage: 'E-Mail address is required.',
  },
  placeholder: {
    id: `${scope}.placeholder`,
    defaultMessage: 'Enter e-mail address',
  },
});
