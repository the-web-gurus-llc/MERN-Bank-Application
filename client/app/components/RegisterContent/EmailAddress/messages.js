/*
 * EmailAddressForm Messages
 *
 * This contains all the text for the EmailAddressForm component.
 */

import { defineMessages } from 'react-intl';

export const scope = 'app.components.RegisterForm.RegisterContent.EmailAddress';

export default defineMessages({
  label: {
    id: `${scope}.label`,
    defaultMessage: 'E-Mail Address',
  },
  validation_processing_personal_data: {
    id: `${scope}.validation_processing_personal_data`,
    defaultMessage:
      'Confirmation of consent to the processing of personal data is required.',
  },
  validation_email_valid: {
    id: `${scope}.validation_email_valid`,
    defaultMessage: 'E-Mail address must be valid.',
  },
  validation_email_required: {
    id: `${scope}.validation_email_required`,
    defaultMessage: 'E-Mail address is required.',
  },
  placeholder: {
    id: `${scope}.placeholder`,
    defaultMessage: 'Enter e-mail address',
  },
  checkbox_content: {
    id: `${scope}.checkbox_content`,
    defaultMessage: 'I agree to the processing of my personal data.',
  },
  information: {
    id: `${scope}.information`,
    defaultMessage: 'Registration does not require confirmation by the email.',
  },
  emailExist: {
    id: `${scope}.emailExist`,
    defaultMessage: 'E-Mail address already exists.',
  },
});
