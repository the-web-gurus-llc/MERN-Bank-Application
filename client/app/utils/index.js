export { default as colors } from './colors';
export { default as media } from './media';
export { default as request } from './request';
export { default as typography } from './typography';
export { default as api } from './api';
export { default as routes } from './routes';
export { default as locales } from './locales';
export { default as i18n } from './i18n';
