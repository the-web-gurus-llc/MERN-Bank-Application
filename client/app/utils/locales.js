exports.DEFAULT_LOCALE = 'en';
exports.DEFAULT_DATE_FORMAT = 'dd.MM.yyyy, hh:mm aa';

// prettier-ignore
exports.appLocales = [
  'en',
  'de',
  'pl',
];
