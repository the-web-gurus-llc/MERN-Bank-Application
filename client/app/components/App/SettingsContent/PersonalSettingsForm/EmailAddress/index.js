import React from 'react';
import { StyledFormItem } from 'components/Form/styles';
import { useDispatch, useSelector } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { makeSelectUser } from 'containers/SettingsPage/selectors';
import { changeInputAction, checkEmailAction } from 'containers/App/actions';
import { Input } from 'antd';
import { injectIntl, intlShape } from 'react-intl';
import { disabledSpacesInput } from 'helpers';
import PropTypes from 'prop-types';
import messages from './messages';

const stateSelector = createStructuredSelector({
  user: makeSelectUser(),
});

function EmailAddress({ intl, onValidateFields }) {
  const dispatch = useDispatch();
  const { user } = useSelector(stateSelector);

  const onCheckEmail = (value, reject, resolve) =>
    dispatch(checkEmailAction(value, reject, resolve));
  const onChangeInput = (event) => dispatch(changeInputAction(event.target));

  const checkEmailAddressAlreadyExist = (_, value, callback) =>
    new Promise((resolve, reject) => {
      if (value !== user.email) {
        onCheckEmail(value, reject, resolve);
      } else {
        resolve();
      }
    }).catch(() => callback(intl.formatMessage(messages.emailExist)));

  return (
    <StyledFormItem
      label={intl.formatMessage(messages.label)}
      name="email"
      tailed="true"
      hasFeedback
      rules={[
        {
          type: 'email',
          message: intl.formatMessage(messages.validationEmailValid),
        },
        {
          asyncValidator: checkEmailAddressAlreadyExist,
        },
      ]}
    >
      <Input
        onPressEnter={onValidateFields}
        onKeyPress={disabledSpacesInput}
        onChange={(event) => onChangeInput(event)}
        name="email"
        placeholder={intl.formatMessage(messages.placeholder)}
      />
    </StyledFormItem>
  );
}

EmailAddress.propTypes = {
  intl: intlShape.isRequired,
  onValidateFields: PropTypes.func.isRequired,
};

export default injectIntl(EmailAddress);
