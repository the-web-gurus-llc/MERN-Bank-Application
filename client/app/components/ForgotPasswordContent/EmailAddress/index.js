import React from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { Input, Checkbox } from 'antd';
import { intlShape, injectIntl, FormattedMessage } from 'react-intl';
import { checkEmailAction, changeInputAction } from 'containers/App/actions';
import { makeSelectEmail } from 'containers/ForgetPasswordPage/selectors';

import { StyledFormItem } from 'components/Form/styles';
import PropTypes from 'prop-types';
import { disabledSpacesInput } from 'helpers';
import messages from './messages';

const stateSelector = createStructuredSelector({
  email: makeSelectEmail(),
});

function EmailAddress({ intl, onValidateFields }) {
  const { email } = useSelector(stateSelector);
  const dispatch = useDispatch();

  const onChangeInput = (event) => dispatch(changeInputAction(event.target));
 

  return (
    <>
      <StyledFormItem
        label={intl.formatMessage(messages.label)}
        name="email"
        rules={[
          {
            type: 'email',
            message: intl.formatMessage(messages.validation_email_valid),
          },
          {
            required: true,
            message: intl.formatMessage(messages.validation_email_required),
          },
        ]}
      >
        <Input
          onPressEnter={onValidateFields}
          onKeyPress={disabledSpacesInput}
          onChange={(event) => onChangeInput(event)}
          name="email"
          value={email}
          placeholder={intl.formatMessage(messages.placeholder)}
        />
      </StyledFormItem>
    </>
  );
}

EmailAddress.propTypes = {
  intl: intlShape.isRequired,
  onValidateFields: PropTypes.func.isRequired,
};

export default injectIntl(EmailAddress);
