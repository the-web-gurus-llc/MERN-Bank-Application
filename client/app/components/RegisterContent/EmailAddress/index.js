import React from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { Input, Checkbox } from 'antd';
import { intlShape, injectIntl, FormattedMessage } from 'react-intl';
import { checkEmailAction, changeInputAction } from 'containers/App/actions';
import { makeSelectEmail } from 'containers/RegisterPage/selectors';
import { StyledInformation } from 'components/RegisterForm/styles';
import { StyledFormItem } from 'components/Form/styles';
import PropTypes from 'prop-types';
import { disabledSpacesInput } from 'helpers';
import messages from './messages';

const stateSelector = createStructuredSelector({
  email: makeSelectEmail(),
});

function EmailAddress({ intl, onValidateFields }) {
  const { email } = useSelector(stateSelector);
  const dispatch = useDispatch();

  const onChangeInput = (event) => dispatch(changeInputAction(event.target));
  const onCheckEmail = (value, reject, resolve) =>
    dispatch(checkEmailAction(value, reject, resolve));

  const checkEmailAddressAlreadyExist = (_, value, callback) =>
    new Promise((resolve, reject) =>
      onCheckEmail(value, reject, resolve),
    ).catch(() => callback(intl.formatMessage(messages.emailExist)));

  const checkDataProcessingIsAccepted = (_, value) => {
    if (value) {
      return Promise.resolve();
    }

    return Promise.reject(
      new Error(
        intl.formatMessage(messages.validation_processing_personal_data),
      ),
    );
  };

  return (
    <>
      <StyledFormItem
        label={intl.formatMessage(messages.label)}
        name="email"
        hasFeedback
        rules={[
          {
            type: 'email',
            message: intl.formatMessage(messages.validation_email_valid),
          },
          {
            required: true,
            message: intl.formatMessage(messages.validation_email_required),
          },
          { asyncValidator: checkEmailAddressAlreadyExist },
        ]}
      >
        <Input
          onPressEnter={onValidateFields}
          onKeyPress={disabledSpacesInput}
          onChange={(event) => onChangeInput(event)}
          name="email"
          value={email}
          placeholder={intl.formatMessage(messages.placeholder)}
        />
      </StyledFormItem>

      <StyledFormItem
        tailed="true"
        name="confirm-personal-data"
        valuePropName="checked"
        rules={[{ validator: checkDataProcessingIsAccepted }]}
      >
        <Checkbox>
          <FormattedMessage {...messages.checkbox_content} />
        </Checkbox>
      </StyledFormItem>

      <StyledInformation>
        <FormattedMessage {...messages.information} />
      </StyledInformation>
    </>
  );
}

EmailAddress.propTypes = {
  intl: intlShape.isRequired,
  onValidateFields: PropTypes.func.isRequired,
};

export default injectIntl(EmailAddress);
