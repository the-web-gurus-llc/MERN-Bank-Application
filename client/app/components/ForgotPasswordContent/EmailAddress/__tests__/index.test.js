import React from 'react';
import { render } from '@testing-library/react';
import 'utils/__tests__/__mocks__/matchMedia';
import { createMemoryHistory } from 'history';
import { Provider } from 'react-redux';
import { Form } from 'antd';
import { IntlProvider } from 'react-intl';
import { DEFAULT_LOCALE } from 'utils/locales';
import configureStore from 'utils/configureStore';
import EmailAddress from '../index';

describe('<EmailAddress />', () => {
  const history = createMemoryHistory();
  const store = configureStore({}, history);
  const onValidateFields = () => null;

  it('should render a information', () => {
    const { container } = render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <Form>
            <EmailAddress onValidateFields={onValidateFields} />
          </Form>
        </IntlProvider>
      </Provider>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
