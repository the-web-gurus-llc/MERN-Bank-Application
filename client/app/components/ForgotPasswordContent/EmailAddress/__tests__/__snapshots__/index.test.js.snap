// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmailAddress /> should render a information 1`] = `
<form
  class="ant-form ant-form-horizontal"
>
  <div
    class="ant-row ant-form-item Formstyle__StyledFormItem-sc-7x4ylh-6 hnNRbx"
  >
    <div
      class="ant-col ant-form-item-label"
    >
      <label
        class="ant-form-item-required"
        for="email"
        title="E-Mail Address"
      >
        E-Mail Address
      </label>
    </div>
    <div
      class="ant-col ant-form-item-control"
    >
      <div
        class="ant-form-item-control-input"
      >
        <div
          class="ant-form-item-control-input-content"
        >
          <input
            class="ant-input"
            id="email"
            name="email"
            placeholder="Enter e-mail address"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="ant-row ant-form-item Formstyle__StyledFormItem-sc-7x4ylh-6 cHmqRL"
    tailed="true"
  >
    <div
      class="ant-col ant-form-item-control"
    >
      <div
        class="ant-form-item-control-input"
      >
        <div
          class="ant-form-item-control-input-content"
        >
          <label
            class="ant-checkbox-wrapper"
          >
            <span
              class="ant-checkbox"
            >
              <input
                class="ant-checkbox-input"
                id="confirm-personal-data"
                type="checkbox"
                value=""
              />
              <span
                class="ant-checkbox-inner"
              />
            </span>
            <span>
              <span>
                I agree to the processing of my personal data.
              </span>
            </span>
          </label>
        </div>
      </div>
    </div>
  </div>
  <div
    class="RegisterFormstyle__StyledInformation-fqluwk-0 gxwSj"
  >
    <span>
      Registration does not require confirmation by the email.
    </span>
  </div>
</form>
`;
